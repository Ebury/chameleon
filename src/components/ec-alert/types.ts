export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
