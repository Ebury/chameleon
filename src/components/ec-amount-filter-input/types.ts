import type { DropdownItem } from '../ec-dropdown/types';

export interface ComparisonSymbolItem extends DropdownItem<string> {}

export interface AmountFilterInputModel {
  amount?: number,
  comparisonSymbol?: ComparisonSymbolItem,
}

export interface AmountFilterInputProps {
  modelValue: AmountFilterInputModel,
  locale?: string,
  isSensitive?: boolean,
  label?: string,
  note?: string,
  amountPlaceholder?: string,
  bottomNote?: string,
  isWarning?: boolean,
  warningTooltipMessage?: string,
  errorMessage?: string,
  errorTooltipMessage?: string,
  comparisonSymbolItems: ComparisonSymbolItem[],
}
