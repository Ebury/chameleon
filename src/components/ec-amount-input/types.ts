import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
