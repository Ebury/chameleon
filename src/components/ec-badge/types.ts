import { AlertType as BadgeType } from '../ec-alert/types';

export { BadgeType };

export interface BadgeProps {
  type?: BadgeType,
  value: string | number,
}
