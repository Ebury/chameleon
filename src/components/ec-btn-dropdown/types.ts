import type { AnchorHTMLAttributes } from 'vue';
import type { RouteLocationRaw } from 'vue-router';

import type { DropdownItem } from '../ec-dropdown-search/types';

export interface BtnDropdownItem<TValue = string> extends DropdownItem<TValue> {
  href?: string,
  to?: string | RouteLocationRaw,
  attrs?: AnchorHTMLAttributes,
}

export interface BtnDropdownProps<TValue = string, TBtnDropdownItem extends BtnDropdownItem<TValue> = BtnDropdownItem<TValue>> {
  items?: TBtnDropdownItem[],
  isBtnDropdownDisabled?: boolean,
  buttonText: string,
  listDataTest?: string,
  href?: string,
  to?: string | RouteLocationRaw,
}
