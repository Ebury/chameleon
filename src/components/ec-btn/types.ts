import type { RouteLocationRaw } from 'vue-router';

import type { IconName } from '../ec-icon/iconNames';

export enum ButtonSize {
  Medium = 'md',
  Small = 'sm'
}

export enum ButtonCategory {
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Error = 'error',
  Warning = 'warning'
}

export interface ButtonProps {
  size?: ButtonSize;
  href?: string;
  to?: RouteLocationRaw;
  tag?: string;
  isDisabled?: boolean;
  icon?: IconName;
  isRounded?: boolean;
  isOutline?: boolean;
  isFullWidth?: boolean;
  category: ButtonCategory;
  isReverse?: boolean;
  isLoading?: boolean;
  isSubmit?: boolean
}
