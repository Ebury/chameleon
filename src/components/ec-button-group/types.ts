export interface ButtonGroupItem<TValue> {
  id?: string | number,
  value: TValue,
  text: string,
  disabled?: boolean,
}

export interface ButtonGroupProps<TValue> {
  modelValue?: TValue,
  items: ButtonGroupItem<TValue>[],
}
