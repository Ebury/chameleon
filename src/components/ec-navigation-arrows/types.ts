export enum NavigationArrowsEvent {
  RIGHT_ARROW_CLICK = 'right-arrow-click',
  LEFT_ARROW_CLICK = 'left-arrow-click',
}

export interface NavigationArrowsEvents {
  [NavigationArrowsEvent.RIGHT_ARROW_CLICK]: undefined
  [NavigationArrowsEvent.LEFT_ARROW_CLICK]: undefined
}

export interface NavigationArrowsProps {
  isRightArrowDisabled?: boolean
  isLeftArrowDisabled?: boolean
}
