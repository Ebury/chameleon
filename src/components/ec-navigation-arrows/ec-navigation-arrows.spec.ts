import { mount } from '@vue/test-utils';

import type { CVueWrapper } from '../../../tests/utils/global';
import EcNavigationArrows from './ec-navigation-arrows.vue';
import type { NavigationArrowsProps } from './types';
import { NavigationArrowsEvent } from './types';

describe('EcNavigationArrows', () => {
  function mountComponent(props?: NavigationArrowsProps): CVueWrapper {
    return mount(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      EcNavigationArrows as any,
      {
        props,
      },
    ) as unknown as CVueWrapper;
  }

  it('should render properly with both arrows enabled', () => {
    const wrapper = mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render properly with both arrows disabled', () => {
    const wrapper = mountComponent({
      isPreviousDisabled: true,
      isNextDisabled: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should emit an event when click next arrow', () => {
    const wrapper = mountComponent();

    wrapper.findByDataTest('ec-navigation-arrows__next')
      .trigger('click');

    expect(wrapper.emitted(NavigationArrowsEvent.NEXT_CLICK)?.length).toBe(1);
  });

  it('should emit an event when click previous arrow', () => {
    const wrapper = mountComponent();

    wrapper.findByDataTest('ec-navigation-arrows__previous')
      .trigger('click');

    expect(wrapper.emitted(NavigationArrowsEvent.PREVIOUS_CLICK)?.length).toBe(1);
  });

  it('should not emit an event when click on disabled buttons', () => {
    const wrapper = mountComponent({
      isPreviousDisabled: true,
      isNextDisabled: true,
    });

    wrapper.findByDataTest('ec-navigation-arrows__next')
      .trigger('click');
    wrapper.findByDataTest('ec-navigation-arrows__previous')
      .trigger('click');

    expect(wrapper.emitted(NavigationArrowsEvent.PREVIOUS_CLICK)?.length).toBe(undefined);
    expect(wrapper.emitted(NavigationArrowsEvent.NEXT_CLICK)?.length).toBe(undefined);
  });
});
