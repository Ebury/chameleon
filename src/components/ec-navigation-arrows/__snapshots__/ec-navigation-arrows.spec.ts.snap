// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcNavigationArrows should render properly with both arrows disabled 1`] = `
<div
  class="ec-navigation-arrows"
  data-test="ec-navigation-arrows"
>
  <button
    class="ec-navigation-arrows__button ec-navigation-arrows__button--disabled"
    data-test="ec-navigation-arrows__previous"
    disabled=""
    type="button"
  >
    <svg
      class="ec-icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-chevron-left"
      />
    </svg>
  </button>
  <button
    class="ec-navigation-arrows__button ec-navigation-arrows__button--disabled"
    data-test="ec-navigation-arrows__next"
    disabled=""
    type="button"
  >
    <svg
      class="ec-icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-chevron-right"
      />
    </svg>
  </button>
</div>
`;

exports[`EcNavigationArrows should render properly with both arrows enabled 1`] = `
<div
  class="ec-navigation-arrows"
  data-test="ec-navigation-arrows"
>
  <button
    class="ec-navigation-arrows__button"
    data-test="ec-navigation-arrows__previous"
    type="button"
  >
    <svg
      class="ec-icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-chevron-left"
      />
    </svg>
  </button>
  <button
    class="ec-navigation-arrows__button"
    data-test="ec-navigation-arrows__next"
    type="button"
  >
    <svg
      class="ec-icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-chevron-right"
      />
    </svg>
  </button>
</div>
`;
